model-builder/src/builder.rs

```diff
@@ -172,6 +241,29 @@ impl IntoTime for &str {
     }
 }
 
+pub trait IntoDate {
+    fn into_date(self) -> Date;
+}
+
+impl IntoDate for Date {
+    fn into_date(self) -> Date {
+        self
+    }
+}
+
+impl IntoDate for &Date {
+    fn into_date(self) -> Date {
+        *self
+    }
+}
+
+impl IntoDate for &str {
+    // Note: if the string is not in the right format, this conversion will fail
```

Comment:
I don't really know, it's a try only in the builder, and it's handy not to care about the fail.
The alternative is a `TryInto` and `d.try_into().unwrap()` in the `calendar()` method no?

Reply:
That wouldn't solve the problem @pbougue is trying to solve, the hidden trap if we were to use this code anywhere else than for tests. I guess the _panic_ in tests is fine, but we need to take care to these kinds of things if we were to start using it in production code... (note of the insisting guy, it would be obvious if it was named `TestModelBuilder` :wink: ).

Could we add this `IntoDate` trait with a `#[cfg(test)]` clause before (with comments about why), so it doesn't exist unless it's tests? I guess that would protect us? I would make `.calendar()` unusable in production code for now but that the goal, right?

Reply:
how about only protecting the `impl IntoDate for &str`? I'm not sure about hiding the whole `ModelBuilder` behind a `cfg(test)` (and I don't know how it will affect the use of the model builder outside of transit-model :thinking: 

Just for you information, there is the same `panic` for `IntoTime` used by `.st("stop1", "10:10:10", "10:10:11")`

Reply:
I was talking of hiding only the trait `IntoDate` but hiding only the implementation for `&str` should be enough indeed! :+1:

Reply:
grumpf, does not work out of the box, are you sure it's important :roll_eyes: :smile: ?

Reply:
the other way would be to try using `const` expr on this to fail at build time, but it does not seems trivial :/